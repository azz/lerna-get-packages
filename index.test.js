const lernaGetPackages = require(".");

expect.addSnapshotSerializer({
  test: value =>
    typeof value === "string" &&
    (value.indexOf("\\") > -1 || value.indexOf(process.cwd()) > -1),
  print: (value, serializer) =>
    serializer(value.replace(process.cwd(), "<cwd>").replace(/\\/g, "/"))
});

describe("lernaGetPackages()", () => {
  test("matches snapshot", () => {
    const packages = lernaGetPackages(__dirname + "/fixture");
    expect(packages).toMatchSnapshot();
  });

  test("returns empty array if no packages", () => {
    expect(lernaGetPackages(__dirname)).toEqual([]);
  });

  test("returns empty array if path doesn't exist", () => {
    expect(lernaGetPackages("/blargh")).toEqual([]);
  });
});
