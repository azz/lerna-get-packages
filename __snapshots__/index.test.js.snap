// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lernaGetPackages() matches snapshot 1`] = `
Array [
  Object {
    "location": "<cwd>/fixture/packages/bar",
    "package": Object {
      "dependencies": Object {
        "@azz/baz": "*",
      },
      "name": "@azz/bar",
      "version": "0.0.0",
    },
  },
  Object {
    "location": "<cwd>/fixture/packages/baz",
    "package": Object {
      "name": "@azz/baz",
      "version": "0.0.0",
    },
  },
  Object {
    "location": "<cwd>/fixture/packages/foo",
    "package": Object {
      "dependencies": Object {
        "@azz/bar": "*",
      },
      "name": "@azz/foo",
    },
  },
]
`;
